### PR TITLE
feat(validation): reject Ingress, Service and KongConsumer with many plugins of the same type attached

### PR DESCRIPTION
**Pull request by programmer04**
**What this PR does / why we need it**:

[KongPlugins can be attached to](https://docs.konghq.com/kubernetes-ingress-controller/latest/concepts/custom-resources/#kongplugin)
- `KongConsumer`
- `Ingress`
- `Service`

Introduce validation that prevents attaching many plugins of the same to an `Ingress`, `Service`, or `KongConsumer`, because it leads to invalid configuration for Kong Gateway. 

The validation is only possible when plugins have been already configured (accepted by K8s), for plugins mentioned in annotations but not yet configured it assumes that they're valid.

**Which issue this PR fixes**:

Closes https://github.com/Kong/kubernetes-ingress-controller/issues/5602

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: internal/admission/validation/kongplugin/kongplugin.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+type k8sObjectWithAnnotations interface {
+	GetAnnotations() map[string]string
+	GetNamespace() string
+}
+
+// ValidatePluginUniquenessPerObject validates that only one plugin of a particular type is attached to an object.
+// It can only check when instance of particular plugin has been already created (KongPlugin has been applied before).
+// Otherwise it will not be able to check and will return nil.
+func ValidatePluginUniquenessPerObject(pluginStore kongPluginGetter, obj k8sObjectWithAnnotations) error {
```
[the remark this answers is not in the document]
Comment (programmer04) [reply internal/admission/validation/kongplugin/kongplugin.go:23]: No, I haven't considered it. Thanks for pointing it out. I'll change it. It should be more real-time

[the remark this answers is not in the document]
Comment (programmer04) [reply internal/admission/validation/kongplugin/kongplugin.go]: Reworked as suggested, also I've added `Service` to this PR. PTAL

**Review by czeslavo — APPROVED** [review]

File: internal/admission/validation/ingress/ingress_test.go [hunk 2]
```diff
@@ -56,6 +57,7 @@ func TestValidateIngress(t *testing.T) {
 				tt.ingress,
 				logger,
 				fakestore,
+				fake.NewClientBuilder().Build(),
```
Comment (pmalek) [inline internal/admission/validation/ingress/ingress_test.go:60]: Nit: this seems to be the same as `fake.NewFakeClient()`.

Comment (programmer04) [reply internal/admission/validation/ingress/ingress_test.go:60]: Resolved in https://github.com/Kong/kubernetes-ingress-controller/pull/5979 💪

**Pull request merged**